### ISSUE TITLE
Possibly missing PHP extension in System Requirements

### ISSUE THREAD

**Issue by pardahlman** [akeneo/pim-docs#565]
Hello,
I run in to some problems when trying to preview an uploaded image (an image attribute in a product). The image previews was not rendered, as the request return a `500 Internal Server Error`. Looking through the logs I found this entry

```
[2017-04-24 11:48:52] request.CRITICAL: Uncaught PHP Exception
Imagine\Exception\NotSupportedException: "PHP exif extension is required to use the 
ExifMetadataReader" at /akeneo/vendor/imagine/imagine/lib/Imagine/Image/Metadata
/ExifMetadataReader.php line 25 {"exception":"[object] (Imagine\\Exception\\NotSupportedException(code:
 0): PHP exif extension is required to use the ExifMetadataReader at /akeneo/vendor/imagine/imagine
/lib/Imagine/Image/Metadata/ExifMetadataReader.php:25)"} []
```

Based on the following: _"PHP exif extension is required to use the ExifMetadataReader"_ it was pretty clear that I needed to install `exif` 🙂 . However, that extension is not listed in the [PHP required modules and configuration](https://docs.akeneo.com/1.7/developer_guide/installation/system_requirements/system_requirements.html) section of the system requirements document.

I'm a bit uncertain if this is the right way to go, as I'm currently seeing a different exception 

```
request.CRITICAL: Uncaught PHP Exception Imagine\Exception\RuntimeException: 
"call_user_func_array() expects parameter 1 to be a valid callback, function 'imagejpeg' not found or 
invalid function name" at /akeneo/vendor/imagine/imagine/lib/Imagine/Gd/Image.php line 697
 {"exception":"[object] (Imagine\\Exception\\RuntimeException(code: 2):call_user_func_array() expects 
parameter 1 to be a valid callback, function 'imagejpeg' not found or invalid function name at /akeneo
/vendor/imagine/imagine/lib/Imagine/Gd/Image.php:697, ErrorException(code: 0): call_user_func_array() 
expects parameter 1 to be a valid callback, function 'imagejpeg' not found or invalid function name at 
/akeneo/vendor/imagine/imagine/lib/Imagine/Gd/Image.php:580)"} []
```

Which I'm currently trying to solve with the help of @gplanchat (on slack):

> on EE you also need imagemagick package, I can’t remember if it is required in CE too

That has not resolved my problems, but it might very well be that I'm missing some other system package that is required.

**Comment by mickaelandrieu** [comment]
Hello @pardahlman and thank you for the feedback, can someone from @akeneo/core-team confirm that `exif` is / is not a PHP requirement for CE/EE versions?

**Comment by nidup** [comment]
Looking at the code, it's only used in EE (for the Product Asset Management feature)

**Comment by mickaelandrieu** [comment]
For Image magick, I've contributed it yesterday: https://github.com/akeneo/pim-docs/blob/1.5/developer_guide/installation/system_requirements/system_requirements.rst.inc#L47. Do we need - in addition - `exif` extension for EE ?

@pardahlman I'll be glad to merge your contribution if you add the `exif` extension in the ``System Requirements`` section of our documentation 👍

**Comment by nidup** [comment]
yep exif for EE :)

**Comment by pardahlman** [comment]
Thanks for giving this attention - I really appreciate it 👌 
Do I understand you correctly that:

1. `exif` is only needed for EE
2. `imagemagick` is only needed for EE

I'm running 1.7.2 CE (here's the [system-info](https://github.com/akeneo/pim-docs/files/999233/akeneo-pim-system-info_2017-05-14_12-14.txt)) and the log entries above comes from that instance. To me, the following _"PHP exif extension is required to use the ExifMetadataReader"_ really suggests that `exif` is needed, wouldn't you say? Especially since installing it mitigated the problem (but exposed a different one)

It would be interesting to hear your thought on the second log entry as well

```
[2017-05-14 12:16:17] request.CRITICAL: Uncaught PHP Exception Imagine\Exception\RuntimeException: "call_user_func_array() expects parameter 1 to be a valid callback, function 'imagejpeg' not found or inval
id function name" at /akeneo/vendor/imagine/imagine/lib/Imagine/Gd/Image.php line 697 {"exception":"[object] (Imagine\\Exception\\RuntimeException(code: 2): call_user_func_array() expects parameter 1 to be
a valid callback, function 'imagejpeg' not found or invalid function name at /akeneo/vendor/imagine/imagine/lib/Imagine/Gd/Image.php:697, ErrorException(code: 0): call_user_func_array() expects parameter 1
to be a valid callback, function 'imagejpeg' not found or invalid function name at /akeneo/vendor/imagine/imagine/lib/Imagine/Gd/Image.php:580)"} []
```

If you want to reproduce it yourselves, you can spin up an Akeneo instance using [this docker-commpose](https://github.com/pardahlman/docker-akeneo/blob/master/docker-compose.yml) (here's the file that installs [the PHP extensions](https://github.com/pardahlman/docker-akeneo/blob/master/akeneo/installation/php/install.sh).

Looking forward hearing from you!

**Comment by pardahlman** [comment]
Hello again,

I looked into this again, and it turns out that GD was not compiled with the flag `--with-jpeg-dir=/usr/include/`. After re-compiling with the flag and restarting Apache it works 🎉 

However, I'm pretty certain that `exif` needs to be installed for CE as well. So it should be added to the list of requirements.

**Comment by pardahlman** [comment]
Reading the System Requirements again, it says

> php5-gd 	No specific configuration

Don't know if it makes sense to say that it needs to be configured according to above?

**Comment by dnd-tyler** [comment]
Hi, 
I'll take it :smile: 
So I reproduct the bug on my local. When I upload an image on a product with exif disable, I dont have the preview. And when exif is actived, the preview is here :)
Also, in demo.akeneo.com, the exif extension is enabled too.
So I will add php-exif in system-requirement. But I guess, I have to add the php-exif in the requirement of the package who need It. 
Is it liip/imagine-bundle or imagine/Imagine or other ?

**Comment by mickaelandrieu** [comment]
Hello @dnd-tyler!
It's a very good question: I don't know :) The best we can do is add it to our own `composer.json`: if it's Imagine the issue, as Symfony is welcoming [this library](https://github.com/symfony/symfony/pull/21820), we'll probably rely on it later.

Also you may complete the Pim requirements => https://github.com/akeneo/pim-community-dev/blob/master/app/PimRequirements.php#L53

What do you think?

**Comment by dnd-tyler** [comment]
I opened two MR : 
https://github.com/akeneo/pim-community-dev/pull/6314
https://github.com/akeneo/pim-docs/pull/623

**Comment by damien-carcel** [comment]
Issue has been fixed starting 2.0. Thanks for it everyone.
